packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -414,20 +414,18 @@ class CupertinoDatePicker extends StatefulWidget {
   final double itemExtent;
 
   @override
-  State<StatefulWidget> createState() { // ignore: no_logic_in_create_state, https://github.com/flutter/flutter/issues/70499
+  State<StatefulWidget> createState() {
     // The `time` mode and `dateAndTime` mode of the picker share the time
     // columns, so they are placed together to one state.
     // The `date` mode has different children and is implemented in a different
     // state.
-    switch (mode) {
-      case CupertinoDatePickerMode.time:
-      case CupertinoDatePickerMode.dateAndTime:
-        return _CupertinoDatePickerDateTimeState();
-      case CupertinoDatePickerMode.date:
-        return _CupertinoDatePickerDateState(dateOrder: dateOrder);
-      case CupertinoDatePickerMode.monthYear:
-        return _CupertinoDatePickerMonthYearState(dateOrder: dateOrder);
-    }
+    // ignore: no_logic_in_create_state, https://github.com/flutter/flutter/issues/70499
```

Comment:
This ignore made more sense at its old location. Can it be moved back? Or at the very least above the comment?

Reply:
Unfortunately, it needs to be placed right above the `switch` expression in order to suppress the warning.

But I was able to reposition the other comments; hopefully it's easier to follow now.